### PR TITLE
add sample code

### PR DESCRIPTION
**Pull request by TakaValley**
# Description

Add the sample code for below scenario:
1.Extracting table cell values from analysis result.
2.Sort the paragraphs from analysis result, by the right reading order.

# All SDK Contribution checklist:
- [X] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_layout_and_sort_paragraphs.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+
+"""
+FILE: sample_analyze_custom_documents_and_read_table_cells.py
```
Comment (toothache) [inline sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_layout_and_sort_paragraphs.py]: This does not match the file name

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_layout_and_sort_paragraphs.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+DESCRIPTION:
+    This sample demonstrates how to analyze a document with a prebuilt-read built model
```
Comment (toothache) [inline sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_layout_and_sort_paragraphs.py]: If you use `prebuilt-read` model, why the file is called `analyze_layout_xxxx`?

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_layout_and_sort_paragraphs.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        "prebuilt-read", formUrl
+    )
+
+    result = poller.result()
+    result.paragraphs.sort(key=lambda p: p.spans[0].offset)
+    print("----Verify the order of paragrahps------")
+    for idx, paragraph in enumerate(result.paragraphs):
+        print(f"...paragraph:{idx}, offset:{paragraph.spans[0].offset}")
+
+    print("----------------------------------------")
```
Comment (toothache) [inline sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_layout_and_sort_paragraphs.py]: This only adds a very simple code to enumerate the paragraphs, should we simply extend in `sample_analyze_layout.py`?

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_custom_documents_and_read_table_cells.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        for region in table.bounding_regions:
+            print("...{}".format(i + 1, region.page_number))
+        for cell in table.cells:
+            print(
+                "...Cell[{}][{}] has content '{}'".format(
+                    cell.row_index, cell.column_index, cell.content.encode("utf-8")
+                )
+            )
+
+    print("----------------------------------------")
```
Comment (toothache) [inline sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_custom_documents_and_read_table_cells.py]: Same, should we directly update in `sample_analyze_custom_documents`?

**Review by toothache — CHANGES_REQUESTED** [review]

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_layout_and_sort_paragraphs.py (hunk 2 above)
Comment (TakaValley) [reply sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_layout_and_sort_paragraphs.py]: resolved, thx

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_layout_and_sort_paragraphs.py (hunk 3 above)
Comment (TakaValley) [reply sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_layout_and_sort_paragraphs.py]: moved, thx

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_custom_documents_and_read_table_cells.py (hunk 4 above)
Comment (TakaValley) [reply sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_custom_documents_and_read_table_cells.py]: optimized, thx

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_custom_documents.py [hunk 5]
```diff
@@ -89,6 +89,22 @@ def analyze_custom_documents(custom_model_id):
                     print(f"...{region.page_number}")
             for cell in table.cells:
                 print(f"...Cell[{cell.row_index}][{cell.column_index}] has text '{cell.content}'")
+
+    # Extract table cell values from "documents" field in result and print the name/value information
+    if result.documents:
+        for idx, doc in enumerate(result.documents):
```
Comment (toothache) [inline sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_custom_documents.py]: for idx, doc in enumerate(result.documents or []):

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_custom_documents.py [hunk 6]
```diff
@@ -89,6 +89,22 @@ def analyze_custom_documents(custom_model_id):
...
+            print(f"\n{doc.fields.__len__()} fields found on Document {idx}" )
```
Comment (toothache) [inline sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_custom_documents.py]: `{len(doc.fields)}`

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_custom_documents.py [hunk 7]
```diff
@@ -89,6 +89,22 @@ def analyze_custom_documents(custom_model_id):
...
+            for fieldName,fieldValue in doc.fields.items():
+                # "TableLayout" is the table field name which you labeled. Table cell information store as array in documnet field.
+                if fieldName=="TableLayout" and fieldValue.type=="array" and fieldValue.value_array:
```
Comment (toothache) [inline sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_custom_documents.py]: nit: you need to spare whitespaces before and after `==`. can you enable python auto formatter in vscode and use it to format your code?

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_custom_documents.py [hunk 8]
```diff
@@ -89,6 +89,22 @@ def analyze_custom_documents(custom_model_id):
...
+                    print(f"------Start Extracting Table Cell Values------")
```
Comment (toothache) [inline sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_custom_documents.py]: the `f` prefix is not needed.

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_custom_documents.py [hunk 9]
```diff
@@ -89,6 +89,22 @@ def analyze_custom_documents(custom_model_id):
...
+                    for rowIdx, obj in enumerate(fieldValue.value_array):
+                        valObj= obj["valueObject"]
+                        if valObj:
+                            for key in valObj.keys():
+                                print(f"...RowIndex:{rowIdx}, ColName: {key}, CellValue: {getattr(valObj[key], "content")}")
```
Comment (toothache) [inline sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_custom_documents.py]: can't we use `getattr(valObj[key], "content")` => `valObj[key].content`

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_custom_documents.py [hunk 10]
```diff
@@ -89,6 +89,22 @@ def analyze_custom_documents(custom_model_id):
...
+                    print(f"------End Extracting Table Cell Values------")
```
Comment (toothache) [inline sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_custom_documents.py]: `f` can be omitted

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_read.py [hunk 11]
```diff
@@ -106,6 +106,11 @@ def analyze_read():
         for paragraph in result.paragraphs:
             print(f"Found paragraph with role: '{paragraph.role}' within {paragraph.bounding_regions} bounding region")
             print(f"...with content: '{paragraph.content}'")
+        
+        print(f"----Sort the paragraphs by reading order----")
```
Comment (toothache) [inline sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_read.py]: remove `f`

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_read.py [hunk 12]
```diff
@@ -106,6 +106,11 @@ def analyze_read():
...
+        result.paragraphs.sort(key=lambda p: p.spans[0].offset)
+        for idx, paragraph in enumerate(result.paragraphs):
+            print(f"...paragraph:{idx} with offset: {paragraph.spans[0].offset} and length: {paragraph.spans[0].length}")
```
Comment (toothache) [inline sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_read.py:113]: paragraph can have multiple spans

**Review by toothache — CHANGES_REQUESTED** [review]

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_custom_documents.py (hunk 5 above)
Comment (TakaValley) [reply sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_custom_documents.py]: fixed thx

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_custom_documents.py (hunk 7 above)
Comment (TakaValley) [reply sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_custom_documents.py]: yes, I install Black Formatter which publish by MS.  Will check the config again.

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_custom_documents.py (hunk 6 above)
Comment (TakaValley) [reply sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_custom_documents.py]: fixed

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_read.py (hunk 11 above)
Comment (TakaValley) [reply sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_read.py]: fixed

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_read.py (hunk 12 above)
Comment (TakaValley) [reply sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_read.py:113]: fixed thx

**Comment by toothache** [comment]
Pls also update the async version of samples: `sdk/documentintelligence/azure-ai-documentintelligence/samples/aio`

**Review by toothache — COMMENTED** [review]
Pls avoid formatting the entire document. Prefer to format only the newly introduced codes.

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_custom_documents.py [hunk 13]
```diff
@@ -35,6 +35,7 @@ def analyze_custom_documents(custom_model_id):
         os.path.join(os.path.abspath(__file__), "..", "./sample_forms/forms/Form_1.jpg")
     )
     # [START analyze_custom_documents]
+    from prettytable import PrettyTable
```
Comment (toothache) [inline sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_custom_documents.py]: I'm not sure if this is a good idea to introduce a third-party library. I didn't see other projects using it.

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_custom_documents.py [hunk 14]
```diff
@@ -43,12 +44,16 @@ def analyze_custom_documents(custom_model_id):
     key = os.environ["DOCUMENTINTELLIGENCE_API_KEY"]
     model_id = os.getenv("CUSTOM_BUILT_MODEL_ID", custom_model_id)
 
-    document_intelligence_client = DocumentIntelligenceClient(endpoint=endpoint, credential=AzureKeyCredential(key))
+    document_intelligence_client = DocumentIntelligenceClient(
+        endpoint=endpoint, credential=AzureKeyCredential(key)
+    )
```
Comment (toothache) [inline sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_custom_documents.py]: Pls avoid formatting the previous code. Only format yours.

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_custom_documents.py (hunk 8 above)
Comment (TakaValley) [reply sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_custom_documents.py]: fixed

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_custom_documents.py (hunk 10 above)
Comment (TakaValley) [reply sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_custom_documents.py]: fixed

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_custom_documents.py (hunk 14 above)
Comment (TakaValley) [reply sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_custom_documents.py]: ok, will notice that

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_custom_documents.py (hunk 9 above)
Comment (TakaValley) [reply sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_custom_documents.py]: optimized

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_custom_documents.py (hunk 13 above)
Comment (TakaValley) [reply sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_custom_documents.py]: fixed , thx

**Comment by TakaValley** [comment]
> Pls also update the async version of samples: `sdk/documentintelligence/azure-ai-documentintelligence/samples/aio`

Fixed, thank you for reminding

**Review by toothache — APPROVED** [review]
LGTM

**Pull request merged**